src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -1433,5 +1433,115 @@ private void CheckConfigurationPaths(RavenConfiguration parentConfiguration, Rav
                                                     $"There is an intersection with database {currentDatabaseName} Temp file located in {currentConfiguration.Indexing.TempPath.FullPath}.");
             }
         }
+
+        public static void NotifyFeaturesAboutStateChange(DatabaseRecord record, long index, StateChange state)
+        {
+            if (CanSkipDatabaseRecordChange())
+                return;
+
+            var taken = false;
+            Stopwatch sp = default;
+
+            while (taken == false)
+            {
+                Monitor.TryEnter(state.Locker, TimeSpan.FromSeconds(5), ref taken);
```

Comment:
What is this good for? 
The lock is per state change? Where does this help us?

Reply:
the `state` is per instance of `DocumentDatabase`/`ShardedDatabaseContext` so changes to the same database are synchronized.